### ISSUE TITLE
KVM - Upgrade to main fails if the storage's UUID doesn't follow the UUID format

### ISSUE THREAD

**Issue by slavkap** [apache/cloudstack#6652]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Primary storage, DB
~~~

##### CLOUDSTACK VERSION

~~~
4.17.0.0 to main
~~~

##### SUMMARY
If the UUID of the primary storage doesn't meet the format of the UUID, the upgrade of CloudStack fails. This probably affects the storage plug-ins: Datera/Nexenta/Cloudbyte/StorPool and Local storage if the UUID is set in the `agent.properties` file. This is caused by change #5382 

##### STEPS TO REPRODUCE

~~~
On version 4.16.1.0 or lower
Set UUID to a storage that doesn't meet the format `%-%-%-%`
Upgrade to version 4.17.0.0 or above
~~~

##### EXPECTED RESULTS

~~~
successful upgrade
~~~

##### ACTUAL RESULTS

~~~
java.lang.StringIndexOutOfBoundsException: begin 0, end 16, length 4

or 

java.lang.NumberFormatException: For input string: "SP"
~~~

**Comment by nvazquez** [comment]
@rohityadavcloud @shwstppr it looks like the original issue https://github.com/apache/cloudstack/pull/5382 can be restricted to Vmware only, a quick fix would be extending the SQL query to consider Vmware results only. I've created the PR: https://github.com/apache/cloudstack/pull/6654 on top of 4.17.0.1, would this need a 4.17.0.2 tag?

**Comment by rohityadavcloud** [comment]
nope - we can only attempt to fix this in 4.17.1.0 or 4.18.0.0 - users who can't upgrade need to build their own from the PR or wait for the next official release

**Comment by shwstppr** [comment]
Fixed with #6654 
@slavkap if possible, please test latest night and re-open if needed
